scipy/interpolate/interpnd.pyx

```diff
@@ -73,36 +73,15 @@ class NDInterpolatorBase:
         else:
             self.tri = None
 
+        self.ndim = ndim
         points = _ndim_coords_from_arrays(points)
-        values = np.asarray(values)
 
-        _check_init_shape(points, values, ndim=ndim)
+        self.need_contiguous = need_contiguous
+        self._original_fill_value = fill_value
```

Comment:
This version attaches a bunch of attributes to the instance. Let's avoid this: let's instead add hooks for subclasses to override. Subclasses are free to store whatever they wish on an instance, but the scipy version should not unless there is a _very_ compelling reason.

Reply:
Will fix.

Reply:
I chose to pass these values to the `_set_values` functions, is it OK?